node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -333,11 +307,28 @@ impl PartialOrd for CandidateComparator {
 
 impl Ord for CandidateComparator {
 	fn cmp(&self, other: &Self) -> Ordering {
-		match self.relay_parent_block_number.cmp(&other.relay_parent_block_number) {
-			Ordering::Equal => (),
-			o => return o,
+		return match (self.relay_parent_block_number, other.relay_parent_block_number) {
+			(None, None) => {
+				// No relay parents for both -> compare hashes
+				self.candidate_hash.cmp(&other.candidate_hash)
+			},
+			(Some(self_relay_parent_block_num), Some(other_relay_parent_block_num)) => {
+				match self_relay_parent_block_num.cmp(&other_relay_parent_block_num) {
+					Ordering::Equal => (),
+					o => return o,
+				}
+				// if the relay parent is the same for both -> compare hashes
+				self.candidate_hash.cmp(&other.candidate_hash)
```

Comment:
Given that we already have a nested match, I think this would be easier to understand if that was in the `Ordering::Equal` match arm already.